### ISSUE TITLE
Unable to see channel icons after zap2it downloads them

### ISSUE THREAD

**Issue by fireheadman** [zynine-/HDHRViewerV2.bundle#9]
OSX, El Capitan (v10.11.6)
Plex Media Server (v1.2.2.2857)

Have the following in place:
zap2it account active
zap2it.pl uses config file to pull/create data (provides user/passwd/cache/icon/outfile)
zap2it.pl creates icon dir, cache dir, xmltv.xml file, crontab log (output)
channel-guide-app-0.0.4 running (w/daily restart via health check script)
HDHRViewer.bundle (v0.8.1) put in ~/Library/Application Support/Plex Media Server/plugins/

I see the created icon dir w/plenty of channel icons...
however when I goto Plex->Channels->HDHR Viewer2->All Channels I do not see the icons (only green/white Plex icons)

For HDHR View2 settings I have:
XMLTV Mode: restapi
XMLTV File Location:  points to xmltv.xml location  (this works since I can see episode info)
XMLTV API URL:  default   (since localhost is where it runs)
XMLTV Match by channel: number
Channel Logo match by channel: number

I have played with toggling the "Channel Logo match by channel" to every option and do not see the icons..... What am I missing here?

Thanks 
Ben

**Comment by fireheadman** [comment]
Please let me know if additional information is required to troubleshoot.

**Comment by zynine-** [comment]
Do you want TV station icon or Program icon?

For program icon, use the -I option on zap2xml.

For TV station icon using -i flag, you have to copy it directly into the Plugin Resource Folder with certain formatting.

**Comment by fireheadman** [comment]
thank you for the reply...
I actually wasn't aware of the Program Icon option... I will have to check that out.

I did end up finding the location late last night about the location:
/Library/Application Support/Plex Media Server/plugins/HDHRViewerV2.bundle/Contents/Resources/<logo-n.n.png>

This is not very well documented.
Also might add, I had to search to find the size, which is 240x140px for the logo.  (believe that is correct)....   I ended up searching/downloading custom PNG logos and using GIMP to resize them.

**Comment by fireheadman** [comment]
Looks like -l is not valid for program icons.....  per the usage

Usage: zap2xml.pl [-OPTIONS [-MORE_OPTIONS]] [--] [PROGRAM_ARG1 ...]

The following single-character options are accepted:
    With arguments: -A -c -C -d -E -i -l -J -m -n -N -o -p -P -r -s -S -t -u -Y -Z
    Boolean (without arguments): -? -a -b -D -e -F -g -I -j -L -M -O -q -T -U -w -x -z

Options may be merged together.  -- stops processing of options.
Space is not required between options and their arguments.
  [Now continuing due to backward compatibility and excessive paranoia.
   See 'perldoc Getopt::Std' about $Getopt::Std::STANDARD_HELP_VERSION.]
zap2xml zap2xml@gmail.com (2016-05-21)
  -u <username>
  -p <password>
  -d <# of days> (default = 7)
  -n <# of no-cache days> (from end)   (default = 0)
  -N <# of no-cache days> (from start) (default = 0)
  -s <start day offset> (default = 0)
  -o <output xml filename> (default = "xmltv.xml")
  -c <cacheDirectory> (default = "cache")
  **-l <lang> (default = "en")**
  -i <iconDirectory> (default = don't download channel icons)
  -m <#> = offset program times by # minutes (better to use TZ env var)
  -b = retain website channel order
  -x = output XTVD xml file format (default = XMLTV)
  -w = wait on exit (require keypress before exiting)
  -q = quiet (no status output)
  -r <# of connection retries before failure> (default = 3, max 20)
  -e = hex encode entities (html special characters like accents)
  -E "amp apos quot lt gt" = selectively encode standard XML entities
  -F = output channel names first (rather than "number name")
  -O = use old tv_grab_na style channel ids (C###nnnn.zap2it.com)
  -A "new live" = append " *" to program titles that are "new" and/or "live"
  -M = copy movie_year to empty movie sub-title tags
  -U = UTF-8 encoding (default = "ISO-8859-1")
  -L = output "<live />" tag (not part of xmltv.dtd)
  -T = don't cache files containing programs with "\bTBA\b|To Be Announced" titles
  -P http://proxyhost:port = to use an http proxy
  -C <configuration file> (default = "/Users/piraro/.zap2xmlrc")
  -S <#seconds> = sleep between requests to prevent flooding of server
  -D = include details = 1 extra http request per program!
  -I = include icons (image URLs) - 1 extra http request per program!
  -J <xmltv> = include xmltv file in output
  -Y <lineupId> (if not using username/password)
  -Z <zipcode> (if not using username/password)
  -z = use tvguide.com instead of zap2it.com
  -a = output all channels (not just favorites) on tvguide.com
  -j = add "series" category to all non-movie programs

**Comment by zynine-** [comment]
capital i.

**-I = include icons (image URLs) - 1 extra http request per program!**

**Comment by fireheadman** [comment]
thank you... again.   Always nice to have a second set of eyes.
